AstarteDeviceSDKCSharp/Data/AstarteFailedMessageStorage.cs

```diff
@@ -54,25 +53,27 @@ public async Task InsertStored(string topic, byte[] payload, int qos, Guid guid)
         {
             try
             {
-
-                if (sqliteConnection.State != ConnectionState.Open)
+                lock (_storeLock)
```

Comment:
I'm not sure locking here is right, We should have only a single instance of `AstarteFailedMessageStorage`, this constructor should be called only the first time we initialize the AstarteDeviceSDK, If it is called multiple times covert it to a singleton.

Reply:
AstarteFailedMessageStorage is instanciated only once on device creation.

Reply:
I tried to remove the locking block, but then I got an error (Index out of range) while closing the sqlite command block. If we remove the sqlcommand block, then we are facing a memory leak. So, my suggestion is to leave it as it is.

Reply:
mmm, it is strange maybe the problem is relative to something else, but ok leave it for now.